exporter/jaegerexporter/exporter.go

```diff
@@ -66,24 +64,34 @@ type protoGRPCSender struct {
 
 func (s *protoGRPCSender) pushTraceData(
 	ctx context.Context,
-	td consumerdata.TraceData,
+	td pdata.Traces,
 ) (droppedSpans int, err error) {
 
-	protoBatch, err := jaegertranslator.OCProtoToJaegerProto(td)
+	batches, err := jaegertranslator.InternalTracesToJaegerProto(td)
 	if err != nil {
-		return len(td.Spans), consumererror.Permanent(err)
+		return td.SpanCount(), consumererror.Permanent(err)
 	}
 
 	if s.metadata.Len() > 0 {
 		ctx = metadata.NewOutgoingContext(ctx, s.metadata)
 	}
 
-	_, err = s.client.PostSpans(
-		ctx,
-		&jaegerproto.PostSpansRequest{Batch: *protoBatch})
+	var sentSpans int
+	for _, batch := range batches {
+		_, err = s.client.PostSpans(
+			ctx,
+			&jaegerproto.PostSpansRequest{Batch: *batch})
 
-	if err != nil {
-		droppedSpans = len(protoBatch.Spans)
+		if err == nil {
+			sentSpans += len(batch.Spans)
+		}
+	}
+
+	droppedSpans = td.SpanCount() - sentSpans
+
+	// Do not report an error if any batches were sent
```

Comment:
Why do we not report an error?

Reply:
Because I think it may cause retries and eventually double-publishing of traces to a backend. Or I miss something?

Reply:
We normally consider double-publishing as less evil than data loss.
Also we can return `consumererror.Permanent(err)`, in that case no retries will be made. I am more worried that if we do not return an error the data will be silently dropped.

Reply:
Ok, makes sense. I that case, i'll change it to respond with error right once it's encountered. I'll make double publishing possible but reduce chance of data loss

Reply:
changed